### PR TITLE
Add rustfmt and clippy into CI

### PR DESCRIPTION
**Pull request by ikrivosheev**
Closes: https://github.com/servo/ipc-channel/issues/222

- [x] added rustfmt and clippy into CI
- [x] fixed clippy warnings 
- [x] replaced `actions-rs/toolchain` to `dtolnay/rust-toolchain`

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -16,14 +16,18 @@ jobs:
...
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@nightly
         with:
-          profile: minimal
-          toolchain: nightly
-          override: true
+          components: rustfmt, clippy
+
+      - name: rustfmt
+        run: cargo fmt --features "${{ matrix.features }}"
```
Comment (mrobinson) [inline .github/workflows/main.yml:27]: This should probably run in `check` mode in order to fail on the CI if the input isn't formatted?

**Review by mrobinson — CHANGES_REQUESTED** [review]
Thanks for this change! In order to verify the clippy fixes, would you be willing to split this PR into two? The first could do a rustfmt and then the second could fix the clippy issues. That would make this a bit easier to review. Thanks!

File: .github/workflows/main.yml (hunk 1 above)
Comment (ikrivosheev) [reply .github/workflows/main.yml:27]: @mrobinson hello! thank you for the review, yes, i can.

Comment (ikrivosheev) [reply .github/workflows/main.yml:27]: First PR: https://github.com/servo/ipc-channel/pull/324. After merge it I will create new PR with clippy.

**Pull request closed without merging**